pkg/vault/test/authorizer/TimelockAuthorizerMigrator.test.ts

```diff
@@ -0,0 +1,109 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { Contract } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+import { deploy, deployedAt } from '@balancer-labs/v2-helpers/src/contract';
+import { ZERO_ADDRESS, ZERO_BYTES32 } from '@balancer-labs/v2-helpers/src/constants';
+
+describe('TimelockAuthorizerMigrator', () => {
+  let EVERYWHERE: string, GRANT_ACTION_ID: string, REVOKE_ACTION_ID: string;
+  let user1: SignerWithAddress, user2: SignerWithAddress, user3: SignerWithAddress;
+  let vault: Contract, oldAuthorizer: Contract, newAuthorizer: Contract, migrator: Contract;
+
+  before('set up signers', async () => {
+    [, user1, user2, user3] = await ethers.getSigners();
+  });
+
+  let rolesData: Array<{ role: string; target: string }>;
+  const ROLE_1 = '0x0000000000000000000000000000000000000000000000000000000000000001';
+  const ROLE_2 = '0x0000000000000000000000000000000000000000000000000000000000000002';
+  const ROLE_3 = '0x0000000000000000000000000000000000000000000000000000000000000003';
+
+  sharedBeforeEach('set up vault', async () => {
+    oldAuthorizer = await deploy('MockBasicAuthorizer');
+    vault = await deploy('Vault', { args: [oldAuthorizer.address, ZERO_ADDRESS, 0, 0] });
+  });
+
+  sharedBeforeEach('set up permissions', async () => {
+    const target = await deploy('MockBasicAuthorizer'); // any contract
+    await oldAuthorizer.grantRolesToMany([ROLE_1, ROLE_2, ROLE_3], [user1.address, user2.address, user3.address]);
+    rolesData = [
+      { role: ROLE_1, target: target.address },
+      { role: ROLE_2, target: target.address },
+      { role: ROLE_3, target: target.address },
+    ];
+  });
+
+  sharedBeforeEach('set up migrator', async () => {
+    migrator = await deploy('TimelockAuthorizerMigrator', { args: [vault.address, oldAuthorizer.address, rolesData] });
+    newAuthorizer = await deployedAt('TimelockAuthorizer', await migrator.newAuthorizer());
+    const setAuthorizerActionId = await actionId(vault, 'setAuthorizer');
+    await oldAuthorizer.grantRolesToMany([setAuthorizerActionId], [migrator.address]);
+  });
+
+  sharedBeforeEach('setup constants', async () => {
+    EVERYWHERE = await newAuthorizer.EVERYWHERE();
+    GRANT_ACTION_ID = await newAuthorizer.GRANT_ACTION_ID();
+    REVOKE_ACTION_ID = await newAuthorizer.REVOKE_ACTION_ID();
+  });
+
+  it('runs the migration properly', async () => {
+    expect(await migrator.migratedRoles()).to.be.equal(0);
+
+    await migrator.migrate(0);
```

Comment:
We should also have tests for partial migrations

Reply:
In fact, given that the other tests are replicated in the fork tests, maybe these are the only relevant cases to unit test?